icicle/include/icicle/fields/field.h

```diff
@@ -883,13 +575,19 @@ class Field
     // TODO: change to a more efficient squaring
     return xs * xs;
   }
-
+#ifdef BARRET
   static constexpr HOST_DEVICE_INLINE Field to_montgomery(const Field& xs) { return xs * Field{CONFIG::montgomery_r}; }
-
   static constexpr HOST_DEVICE_INLINE Field from_montgomery(const Field& xs)
   {
     return xs * Field{CONFIG::montgomery_r_inv};
   }
+#else
+  static constexpr HOST_DEVICE_INLINE Field to_montgomery(const Field& xs)
+  {
+    return xs * Field{CONFIG::montgomery_r_sqr};
```

Comment:
why is this code different for montgomery and barret?

Reply:
the conversion is different